### PR TITLE
Remove pk_is_thread_default check in pk_backend_is_eula_valid

### LINKED ISSUE #300: Unable to call pk_backend_is_eula_valid() in backend methods
All the real actions are done in a new thread using pk_backend_job_thread_create() inside zypp backend, when we call pk_backend_is_eula_valid to check the EULA id, it fails saying it's not in the default thread, which is caused by this check.
> g_return_val_if_fail (pk_is_thread_default (), NULL);

This is introduced in 44bd3ab6.

I'd like to hear your recommendation on how to make such a pk_backend_is_eula_valid call. Should we simply disable that check in pk_backend_is_eula_valid?

### PR DESCRIPTION
**Pull request by JonathanKang**
pk_is_thread_default() check was introduced in commit 44bd3ab6 to
detect possible threading problems.

Some backends need call pk_backend_is_eula_valid() in a thread which is
not the default one to perform EULA related actions. This way
pk_is_thread_default() check fails.

Fix that by removing the pk_is_thread_default() check in
pk_backend_is_eula_valid().

Fixes #300

### CODE REVIEW

**Comment by JonathanKang** [comment]
> Is there a mutex protecting priv->eulas?

Emm, right. There is not mutex lock protecting priv->eulas in pk-backend.c. We should use mutex lock to protect priv->eulas then.

**Comment by hughsie** [comment]
> We should use mutex lock to protect priv->eulas then

Right. A new mutex would be fine.

**Comment by JonathanKang** [comment]
Please review this if you have some free time. Thanks.

File: src/pk-backend.c [hunk 1]
```diff
@@ -955,9 +958,10 @@ pk_backend_is_eula_valid (PkBackend *backend, const gchar *eula_id)
 
 	g_return_val_if_fail (PK_IS_BACKEND (backend), FALSE);
 	g_return_val_if_fail (eula_id != NULL, FALSE);
-	g_return_val_if_fail (pk_is_thread_default (), FALSE);
 
+	g_mutex_lock (&backend->priv->eulas_mutex);
```
Comment (hughsie) [inline src/pk-backend.c]: Could we use GMutexLocker for all these cases?

**Pull request merged**